File: server/datasets/python_source-code/Multiple-A2/organic8_bubble_sort.py
"""
Bubble Sort - organic8
Approach: Classic nested for loops but tracks and prints a running
swap count, and writes the comparison as `not (a <= b)` instead of
`a > b`.
"""


def bubble_sort(arr):
    n = len(arr)
    swap_count = 0

    for i in range(n - 1):
        for j in range(n - 1 - i):
            a = arr[j]
            b = arr[j + 1]
            if not (a <= b):
                arr[j], arr[j + 1] = b, a
                swap_count += 1

    print(f"Total swaps: {swap_count}")
    return arr


if __name__ == "__main__":
    ARR = [64, 34, 25, 12, 22, 11, 90, 5, 77, 1]
    print("Final:", bubble_sort(ARR))

File: server/datasets/java_source-code/Multiple-A2/organic8_bubble_sort.java
/*
 * Bubble Sort - organic8
 * Approach: Classic nested for loops tracking swap count and writing
 * comparison as !(a <= b) instead of a > b.
 */

import java.util.Arrays;

class BubbleSort {
    public static int[] bubbleSort(int[] arr) {
        int n = arr.length;
        int swapCount = 0;

        for (int i = 0; i < n - 1; i++) {
            for (int j = 0; j < n - 1 - i; j++) {
                int a = arr[j];
                int b = arr[j + 1];
                if (!(a <= b)) {
                    arr[j] = b;
                    arr[j + 1] = a;
                    swapCount++;
                }
            }
        }

        System.out.println("Total swaps: " + swapCount);
        return arr;
    }

    public static void main(String[] args) {
        int[] arr = {64, 34, 25, 12, 22, 11, 90, 5, 77, 1};
        System.out.println("Final: " + Arrays.toString(bubbleSort(arr)));
    }
}
